src/sidebar/util/fetch-config.js

```diff
@@ -75,3 +97,87 @@ export function fetchConfig(appConfig, window_ = window) {
     return mergedConfig;
   });
 }
+
+/**
+ * Merge client configuration from h service with config fetched from
+ * an embedding frame synchronously.
+ *
+ * Use this method to retrieve the config synchronously from the embedding
+ * frame's query string. See tests for more details.
+ *
+ * @param {Object} appConfig - App config settings rendered into `app.html` by the h service.
+ * @param {Object} hostPageConfig - App configuration specified by the embedding  frame.
+ * @return {Object} - The merged settings.
+ */
+function fetchConfigEmbed(appConfig, hostPageConfig) {
+  const mergedConfig = {
+    ...appConfig,
+    ...hostPageConfig,
+  };
+  mergedConfig.apiUrl = getApiUrl(mergedConfig);
+  return mergedConfig;
+}
+
+/**
+ * Merge client configuration from h service with config fetched from
+ * an embedding frame asynchronously.
+ *
+ * Use this method to retrieve the config asynchronously from an ancestor
+ * frame via RPC. See tests for more details.
+ *
+ * @param {Object} appConfig - Settings rendered into `app.html` by the h service.
+ * @param {Window} parentFrame - Frame to send call to.
+ * @param {string} origin - Origin filter for `window.postMessage` call.
+ * @return {Promise<Object>} - The merged settings.
+ */
+async function fetchConfigRpc(appConfig, parentFrame, origin) {
+  const remoteConfig = await postMessageJsonRpc.call(
+    parentFrame,
+    origin,
+    'requestConfig',
+    [],
+    3000
+  );
+  return fetchConfigEmbed(appConfig, remoteConfig);
+}
+
+/**
+ * Fetch the host configuration and merge it with the app configuration from h.
+ *
+ * There are 3 ways to get the host config:
+ *  Direct embed - From the hash string of the embedder frame.
+ *  Legacy RPC with unknown parent - From a ancestor parent frame that passes it down via RPC. (deprecated)
+ *  RPC with known parent - From a ancestor parent frame that passes it down via RPC.
+ *
+ * @param {Object} appConfig - Settings rendered into `app.html` by the h service.
+ * @param {Window} window_ - Test seam.
+ * @return {Object} - The merged settings.
+ */
+export async function fetchConfig(appConfig, window_ = window) {
+  const hostPageConfig = hostConfig(window);
+  const requestConfigFromFrame = hostPageConfig.requestConfigFromFrame;
+
+  if (!requestConfigFromFrame) {
+    // Directly embed: just get the config.
+    return fetchConfigEmbed(appConfig, hostPageConfig);
+  }
+  if (typeof requestConfigFromFrame === 'string') {
+    // Legacy: send RPC to all parents to find config. (note will cause js error(s))
```

Comment:
Can you expand on this "will cause js error(s)" comment for the benefit of other developers in future? Something like "Browsers may display errors in the console when messages are  sent to frames that don't match the origin filter".